### PR TITLE
p5js typography + structure + math

### PR DESCRIPTION
**Pull request by anthonygalea**

### CODE REVIEW

File: src/cljc/quil/core.cljc [hunk 1]
```diff
@@ -4048,9 +4052,10 @@
   ([^String s x y]
    (when (current-fill)
      (.text (current-graphics) s (float x) (float y))))
-  ([^String s x y z]
-   (when (current-fill)
-     (.text (current-graphics) s (float x) (float y) (float z))))
+  #?(:clj
```
Comment (nbeloglazov) [inline src/cljc/quil/core.cljc]: Can you update comment to say that 'z' param is available only in clj version?

File: src/cljc/quil/core.cljc [hunk 2]
```diff
@@ -4143,8 +4148,8 @@
   sketches and PDF output in cases where the vector data is available:
   when the font is still installed, or the font is created via the
   create-font fn"
-  ([^PFont font] (.textFont (current-graphics) font))
-  ([^PFont font size] (.textFont (current-graphics) font (int size))))
+  ([font] (.textFont (current-graphics) font))
+  ([font size] (.textFont (current-graphics) font (int size))))
```
Comment (nbeloglazov) [inline src/cljc/quil/core.cljc]: Why remove ^PFont? These are needed for clj to tell jvm which type is it so it doesn't do unnecessary reflection checks.

**Comment by anthonygalea** [comment]
Added **math** as well.

File: src/cljc/quil/core.cljc (hunk 2 above)
Comment (nbeloglazov) [reply src/cljc/quil/core.cljc]: Yes, it's not performance critical, but I want to keep core.cljc without any reflection warnings. It's simpler to have no warnings then remember which are performance non-critical and which one need to be fixed. 

As for type hints in cljs, I dno't think they mean anything there? At least according to this: https://github.com/clojure/clojurescript-site/blob/master/content/about/differences.adoc#hinting So keeping ^PFont even for cljs should not affect anything. Or you saw some issues with that?

**Comment by nbeloglazov** [comment]
For preload. I'd suggest to not support it for now. Looks like it's not strictly necessary as standard way of loading images should still work (I hope): http://quil.info/api/image/loading-and-displaying#load-image

File: src/cljc/quil/snippets/math/trigonometry.cljc [hunk 3]
```diff
@@ -14,6 +14,22 @@
...
+      "angle-mode"
+      {}
+
+      (q/angle-mode :degrees)
+
+      (q/fill "red")
+      (q/rect 50 50 100 100)
+
+      (comment "rotation will be in degrees")
+      (q/rotate q/TWO-PI)
```
Comment (nbeloglazov) [inline src/cljc/quil/snippets/math/trigonometry.cljc]: If it's in degrees, maybe use something like 45? TWO-PI sounds like radians and might be confusing. Also do you mind adding example with (q/angle-mode :radians) for completeness?

**Comment by nbeloglazov** [comment]
Thanks for the thorough work! How about we resolve comments here and I merge this PR? Don't want it to grow very big.

File: src/cljc/quil/snippets/math/trigonometry.cljc (hunk 3 above)
Comment (anthonygalea) [reply src/cljc/quil/snippets/math/trigonometry.cljc]: 👍

File: src/cljc/quil/core.cljc (hunk 2 above)
Comment (anthonygalea) [reply src/cljc/quil/core.cljc]: Didn't know it was ok to leave the type hint for cljs and while adapting the snippet I was under the impression that leaving it resulted in issues. Modified the PR to only contain the change required for the `text-font` snippet. Sorry for the confusion here.

File: src/cljc/quil/core.cljc [hunk 4]
```diff
@@ -1529,7 +1547,8 @@
   completed (or after setup completes if called during the setup
   method). "
   []
-  (.exit (ap/current-applet)))
+  #?(:clj (.exit (ap/current-applet))
+     :cljs (.remove (ap/current-applet))))
```
Comment (nbeloglazov) [inline src/cljc/quil/core.cljc]: I believe this broke cljs test page. We call `(q/exit)` at the end of each snippet: https://github.com/quil/quil/blob/master/test/cljs/quil/snippet.cljs#L37 and now it seems like p5js removes canvas even before we can see it. 

Should we just not call `(q/exit)` after each snippet? Don't know what's the proper way to reuse the same for multiple snippets in p5js.

Comment (anthonygalea) [reply src/cljc/quil/core.cljc]: That's what I had done while [drafting the changes](https://github.com/anthonygalea/quil/commit/e92c6df521d75859fe01ad0b303b8989b019a2a8#diff-8a5b0cf307b160a77d1d44948df43dce). I was just removing any old canvases before a test is run. Works for both single and multiple. What do you think about that approach?

**Comment by anthonygalea** [comment]
> Thanks for the thorough work! How about we resolve comments here and I merge this PR? Don't want it to grow very big.

You're welcome, thank you for the reviewing.

**Pull request merged**